app/controllers/api/v1/application_users_controller.rb

```diff
@@ -92,6 +92,33 @@ def index
   end
 
   ###############################################################
+  # username
+  ###############################################################
+
+  api :GET, '/application_users/username/:username',
+      'Gets a single User with the specified username.'
+  description <<-EOS
+      All users of OpenStax have an associated User object.  This endpoint
+      allows querying additional data on a user when all that is known is the
+      User's username.
+
+      Admins (for Accounts only) are identified by the is_administrator boolean.
+      Some additional user information can be found in associations, such as
+      email addresses in ContactInfos and the password hash in Identity.
+
+      Users have the following String attributes:
+      username, first_name, last_name, full_name, title
+
+    #{json_schema(Api::V1::UserSearchRepresenter, include: :readable)}
+  EOS
+  def username
+    OSU::AccessPolicy.require_action_allowed!(:read, current_api_user, ApplicationUser)
```

Comment:
Looks like `standard_read` won't work since accounts is locked to an older 2.2.4 and it only accepts an id for lookup, not a query or model:  https://github.com/openstax/openstax_api/blob/v2.2.4/lib/openstax/api/roar.rb#L20

I'm wondering if we should return `access_denied` if they don't have permissions though?  That'd allow evil minded applications to determine that a user with that username does exist but they aren't allowed to see it.  

Wouldn't it be better just to return 404?  I've taken that route in the latest commit by using `find!` which will raise `ActiveRecord::RecordNotFound` and be converted to a 404.